[addons] remove the 'void ADDON_Stop()' call

AlwinEsch opened the pull request:
<!--- Provide a general summary of your change in the Title above -->

## Description
Due to support of multiple instances must be (if used) the stop be done
on his instance part.

The Stop is currently used mostly only on "Screensaver" and "Visualisaton"
and added there a own "Stop".

This change need on all addons a update!

<!--- Describe your change in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

AlwinEsch committed:
[addons] remove the 'void ADDON_Stop()' call

Due to support of multiple instances must be (if used) the stop be done
on his instance part.

The Stop is currently used mostly only on "Screensaver" and "Visualisaton"
and added there a own "Stop".

This change need on all addons a update!

ksooo commented on xbmc/cores/VideoPlayer/DVDInputStreams/InputStreamAddon.cpp:
```diff
@@ -31,7 +31,6 @@ CInputStreamAddon::CInputStreamAddon(const CFileItem& fileitem, std::shared_ptr<
 CInputStreamAddon::~CInputStreamAddon()
 {
   Close();
-  m_addon->Stop();
```
I guess you checked that removing this call has no strange side effects?

AlwinEsch replied:
No becomes not used on Inputstream addon's. By my look over the addons is the Stop only used on Screensaver and Visualization.

Currently I go over all addons to check. The pull requests to them comes this time after them is in, are to much to make before.

ksooo replied:
Okay , then.

ksooo reviewed (approved):
Only one question regarding input stream change, otherwise looking good.

AlwinEsch commented:
jenkins build this please

MartijnKaijser commented:
jenkins build this please

ksooo commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h:
```diff
@@ -111,21 +111,21 @@
+#define ADDON_INSTANCE_VERSION_VISUALIZATION          "1.0.1"
+#define ADDON_INSTANCE_VERSION_VISUALIZATION_MIN      "1.0.1"
 #define ADDON_INSTANCE_VERSION_VISUALIZATION_XML_ID   "kodi.binary.instance.visualization"
 #define ADDON_INSTANCE_VERSION_VISUALIZATION_DEPENDS  "xbmc_vis_dll.h" \
```
Not sure, but shouldn't the version numbers for the other add-on types changed in this PR (audio encoder etc.) also be bumped?

ksooo replied:
As discussed on Slack, please update to include the missing bumps.

AlwinEsch committed:
[visualization] change addon interface function table to style of others

Like it was on other types who has used callbacks becomes now also this
changed to the structure style to make it in future easier to add callbacks.

AlwinEsch committed:
[vfs] change addon interface function table to style of
 others

Like it was on other types who has used callbacks becomes now also this
changed to the structure style to make it in future easier to add callbacks.

AlwinEsch committed:
[screensaver] change addon interface function table to style of others

Like it was on other types who has used callbacks becomes now also this
changed to the structure style to make it in future easier to add callbacks.

AlwinEsch committed:
[audioencoder] change addon interface function table to style of others

Like it was on other types who has used callbacks becomes now also this
changed to the structure style to make it in future easier to add callbacks.

AlwinEsch committed:
[image] change addon interface function table to style of others

Like it was on other types who has used callbacks becomes now also this
changed to the structure style to make it in future easier to add callbacks.

AlwinEsch committed:
[audiodecoder] change addon interface function table to style of others

Like it was on other types who has used callbacks becomes now also this
changed to the structure style to make it in future easier to add callbacks.

AlwinEsch commented:
jenkins build this please

ksooo reviewed (approved):
Good to go, imo.

ksooo commented:
jenkins build this please
